### PR TITLE
Rework CMake configuration

### PR DESCRIPTION
**Pull request by kylestach**
## Description
Rework CMake configuration and directory structure. This should speed up incremental builds, as we no longer need to relink the entirety of soccer (and what was previously the robocup.so library) for any change to any of the node.

While we're at it, might as well move soccer/src/soccer -> soccer so we don't have to deal with an extra layer of nested directories for no reason.

## Associated Issue
N/A, I was just annoyed

## Design Documents
N/A

### Do it build?
1. Do it build?

Expected result: It do build.

### CODE REVIEW

File: rj_geometry/CMakeLists.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ add_subdirectory(src)
 # ======================================================================
 # Precompiled headers
 # ======================================================================
-target_precompile_headers(geometry2d PUBLIC include/rj_geometry/point.hpp)
+#target_precompile_headers(geometry2d PUBLIC include/rj_geometry/point.hpp)
```
Comment (kfu02) [inline rj_geometry/CMakeLists.txt:27]: ah this was the bug that messed up my sim too

**Review by kfu02 — APPROVED** [review]
It do build. Love the Annoyance-Driven-Design philosophy you employ.

**Review by kasohrab — COMMENTED** [review]
It didn't build for me. Not sure if it a problem on my end though. The message is long so here's just the top bit:
FAILED: soccer/radio/CMakeFiles/radio.dir/sim_radio.cpp.o 
ccache /usr/bin/ccache /usr/bin/c++  -DDEFAULT_RMW_IMPLEMENTATION=rmw_fastrtps_cpp -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_KEYWORDS -DQT_WIDGETS_LIB -DRCUTILS_ENABLE_FAULT_INJECTION -DSPDLOG_COMPILED_LIB -Isoccer/radio/radio_autogen/include -Isrc/googletest/include -I../rj_common/include -I../rj_geometry/include -I../rj_constants/include -I../rj_convert/include -I../rj_param_utils/include -I../rj_topic_utils/include -I../rj_utils/include -I../rj_config/include -I../soccer -I../external/robocup-fshare/include -isystem /opt/ros/foxy/include -isystem ../install/include -isystem /usr/include/eigen3 -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem rj_protos/include -fdiagnostics-color=always -Werror=return-local-addr -g -g -Og -Werror=return-type -Werror=delete-non-virtual-dtor   -fPIC -std=gnu++17 -MD -MT soccer/radio/CMakeFiles/radio.dir/sim_radio.cpp.o -MF soccer/radio/CMakeFiles/radio.dir/sim_radio.cpp.o.d -o soccer/radio/CMakeFiles/radio.dir/sim_radio.cpp.o -c ../soccer/radio/sim_radio.cpp
In file included from ../install/include/rj_protos/messages_robocup_ssl_wrapper.pb.h:34,
                 from ../install/include/rj_protos/LogFrame.pb.h:36,
                 from ../soccer/debug_drawer.hpp:3,
                 from ../soccer/context.hpp:9,
                 from ../soccer/radio/sim_radio.hpp:10,
                 from ../soccer/radio/sim_radio.cpp:1:
../install/include/rj_protos/messages_robocup_ssl_geometry.pb.h:79:7: error: redefinition of ‘class Vector2f’
   79 | class Vector2f : public ::google::protobuf::Message /* @@protoc_insertion_point(class_definition:Vector2f) */ {
      |       ^~~~~~~~
In file included from rj_protos/include/rj_protos/ssl_simulation_config.pb.h:34,
                 from rj_protos/include/rj_protos/ssl_simulation_control.pb.h:34,
                 from ../soccer/radio/sim_radio.hpp:8,
                 from ../soccer/radio/sim_radio.cpp:1:
rj_protos/include/rj_protos/ssl_vision_geometry.pb.h:123:7: note: previous definition of ‘class Vector2f’
  123 | class Vector2f : public ::google::protobuf::Message /* @@protoc_insertion_point(class_definition:Vector2f) */ {
      |       ^~~~~~~~
In file included from ../install/include/rj_protos/messages_robocup_ssl_wrapper.pb.h:34,
                 from ../install/include/rj_protos/LogFrame.pb.h:36,
                 from ../soccer/debug_drawer.hpp:3,
                 from ../soccer/context.hpp:9,
                 from ../soccer/radio/sim_radio.hpp:10,
                 from ../soccer/radio/sim_radio.cpp:1:
../install/include/rj_protos/messages_robocup_ssl_geometry.pb.h:206:7: error: redefinition of ‘class SSL_FieldLineSegment’
  206 | class SSL_FieldLineSegment : public ::google::protobuf::Message /* @@protoc_insertion_point(class_definition:SSL_FieldLineSegment) */ {
      |       ^~~~~~~~~~~~~~~~~~~~
In file included from rj_protos/include/rj_protos/ssl_simulation_config.pb.h:34,
                 from rj_protos/include/rj_protos/ssl_simulation_control.pb.h:34,
                 from ../soccer/radio/sim_radio.hpp:8,
                 from ../soccer/radio/sim_radio.cpp:1:
rj_protos/include/rj_protos/ssl_vision_geometry.pb.h:250:7: note: previous definition of ‘class SSL_FieldLineSegment’
  250 | class SSL_FieldLineSegment : public ::google::protobuf::Message /* @@protoc_insertion_point(class_definition:SSL_FieldLineSegment) */ {
      |       ^~~~~~~~~~~~~~~~~~~~
In file included from ../install/include/rj_protos/messages_robocup_ssl_wrapper.pb.h:34,
                 from ../install/include/rj_protos/LogFrame.pb.h:36,
                 from ../soccer/debug_drawer.hpp:3,
                 from ../soccer/context.hpp:9,
                 from ../soccer/radio/sim_radio.hpp:10,
                 from ../soccer/radio/sim_radio.cpp:1:

**Review by hdemusg — CHANGES_REQUESTED** [review]
CMake Error at CMakeLists.txt:10 (find_package):
  By not providing "Findament_cmake.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "ament_cmake", but CMake did not find one.

  Could not find a package configuration file provided by "ament_cmake" with
  any of the following names:

    ament_cmakeConfig.cmake
    ament_cmake-config.cmake

  Add the installation prefix of "ament_cmake" to CMAKE_PREFIX_PATH or set
  "ament_cmake_DIR" to a directory containing one of the above files.  If
  "ament_cmake" provides a separate development package or SDK, be sure it
  has been installed.


-- Configuring incomplete, errors occurred!
See also "/home/vagrant/robocup-software/build-debug/rj_geometry_msgs/CMakeFiles/CMakeOutput.log".
ninja: error: loading 'build.ninja': No such file or directory
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:10 (find_package):
  By not providing "Findament_cmake.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "ament_cmake", but CMake did not find one.

  Could not find a package configuration file provided by "ament_cmake" with
  any of the following names:

    ament_cmakeConfig.cmake
    ament_cmake-config.cmake

  Add the installation prefix of "ament_cmake" to CMAKE_PREFIX_PATH or set
  "ament_cmake_DIR" to a directory containing one of the above files.  If
  "ament_cmake" provides a separate development package or SDK, be sure it
  has been installed.


-- Configuring incomplete, errors occurred!
See also "/home/vagrant/robocup-software/build-debug/rj_drawing_msgs/CMakeFiles/CMakeOutput.log".
ninja: error: loading 'build.ninja': No such file or directory
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:10 (find_package):
  By not providing "Findament_cmake.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "ament_cmake", but CMake did not find one.

  Could not find a package configuration file provided by "ament_cmake" with
  any of the following names:

    ament_cmakeConfig.cmake
    ament_cmake-config.cmake

  Add the installation prefix of "ament_cmake" to CMAKE_PREFIX_PATH or set
  "ament_cmake_DIR" to a directory containing one of the above files.  If
  "ament_cmake" provides a separate development package or SDK, be sure it
  has been installed.


-- Configuring incomplete, errors occurred!
See also "/home/vagrant/robocup-software/build-debug/rj_msgs/CMakeFiles/CMakeOutput.log".
ninja: error: loading 'build.ninja': No such file or directory
CMake Error at rj_convert/CMakeLists.txt:10 (find_package):
  By not providing "Findrclcpp.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "rclcpp", but
  CMake did not find one.

  Could not find a package configuration file provided by "rclcpp" with any
  of the following names:

    rclcppConfig.cmake
    rclcpp-config.cmake

  Add the installation prefix of "rclcpp" to CMAKE_PREFIX_PATH or set
  "rclcpp_DIR" to a directory containing one of the above files.  If "rclcpp"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
See also "/home/vagrant/robocup-software/build-debug/CMakeFiles/CMakeOutput.log".
make: *** [makefile:34: all] Error 1

Apologies for the delay. Let me know if this is something I should fix on my end. Thanks!

**Review by hdemusg — APPROVED** [review]

**Comment by hdemusg** [comment]
After fixing things on my side, the sim opened. I assume this is what was desired?

**Comment by kylestach** [comment]
I'll merge after comp

**Pull request closed without merging**